internal/credential/vault/jobs.go

```diff
@@ -270,7 +270,7 @@ func nextRenewal(ctx context.Context, j scheduler.Job) (time.Duration, error) {
 	}
 	defer rows.Close()
 
-	for rows.Next() {
+	if rows.Next() {
```

Comment:
Is this change correct? I am aware that `r.ScanRows` consumes the rows, but isn't it still safer to use a `for` loop?

Reply:
I checked with @louisruch and he confirmed that we should only ever want one row (or zero). The linter had been complaining about the fact that there was an unconditional return in the for loop.

If we need to ScanRows to consume all of them or something we can... @louisruch thoughts here?

Reply:
Oh, I see. Yeah I mean I'd probably change the return to a break?

Reply:
This looks good to me, a break would also work

Reply:
Adding a break suggests that maybe if something goes wrong with the first row that we want to process a second row? But @louisruch said we should really expect only one, which the `if` will guarantee. So personally I'm leaning towards how it is now if that's okay with you two.

Reply:
`if` sounds good to me